src/model.js

```diff
@@ -2825,17 +2825,25 @@ Specify a different name for either index to resolve this issue.`);
         if (options.updateOnDuplicate) {
           options.updateOnDuplicate = options.updateOnDuplicate.map(attr => model.rawAttributes[attr].field || attr);
 
-          const upsertKeys = [];
-
-          for (const i of model.getIndexes()) {
-            if (i.unique && !i.where) { // Don't infer partial indexes
-              upsertKeys.push(...i.fields);
+          if (options.conflictFields) {
+            options.upsertKeys = options.conflictFields.map(
+              attr => (Object.prototype.hasOwnProperty.call(this.rawAttributes, attr)
```

Comment:
This is a pattern we use all the time, we should have a util method `getModelColumnName(model: ModelStatic, attributeName: string): string`

Reply:
Agreed though I think it should be a `protected` attribute of `Model`.  Probably something to be added in a followup pr.